File: src/main/java/net/liyze/basin/core/commands/PublicCommand.java
package net.liyze.basin.core.commands;

import net.liyze.basin.context.annotation.Component;
import net.liyze.basin.core.Command;
import net.liyze.basin.core.CommandParser;
import org.jetbrains.annotations.NotNull;

import java.util.List;

import static net.liyze.basin.core.Main.publicVars;

@Component
public class PublicCommand implements Command {
    @Override
    public void run(@NotNull List<String> args) {
        CommandParser parser = new CommandParser();
        parser.sync().parse(args);
        publicVars.putAll(parser.vars);
    }

    @Override
    public @NotNull String Name() {
        return "public";
    }
}

File: src/main/kotlin/net/liyze/basin/core/scan/PublicCommand.kt
/*
 * Copyright (c) 2023 Liyze09
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.liyze.basin.core.scan

import net.liyze.basin.core.Command
import net.liyze.basin.core.CommandParser
import net.liyze.basin.core.publicVars

class PublicCommand : Command {
    override fun run(args: List<String?>) {
        val parser = CommandParser()
        parser.sync().parse(args.requireNoNulls())
        publicVars.putAll(parser.vars)
    }

    override fun Name(): String {
        return "public"
    }
}
